sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -144,11 +145,16 @@ object AggUtils {
     // [COUNT(DISTINCT foo), MAX(DISTINCT foo)], but [COUNT(DISTINCT bar), COUNT(DISTINCT foo)] is
     // disallowed because those two distinct aggregates have different column expressions.
     val distinctExpressions = functionsWithDistinct.head.aggregateFunction.children
-    val namedDistinctExpressions = distinctExpressions.map {
-      case ne: NamedExpression => ne
-      case other => Alias(other, other.toString)()
+    val normalizedNamedDistinctExpressions = distinctExpressions.map { e =>
```

Comment:
@viirya, out of curiosity, can't we do this in the same place as https://github.com/apache/spark/blob/17586f9ed25cbb1dd7dbb221aecb8749623e2175/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala#L327-L334?

Reply:
Ah, NVM. `distinctExpressions` alone should be normalized here.